contracts/contracts/bridge/HomeBridge.sol

```diff
@@ -90,18 +90,35 @@ contract HomeBridge {
         }
     }
 
-    function _checkSufficientNumberOfConfirmations(uint16 numConfirmations)
+    function purgeConfirmationsFromExValidators(bytes32 transferStateId)
         internal
-        view
-        returns (bool)
     {
-        uint numRequired = (
-                validatorProxy.numberOfValidators() *
-                    validatorsRequiredPercent +
-                    99
-            ) /
-            100;
-        return numConfirmations >= numRequired;
+        address payable[] storage confirmingValidators = transferState[transferStateId]
+            .confirmingValidators;
+
+        uint i = 0;
+        while (i < confirmingValidators.length) {
+            if (validatorProxy.isValidator(confirmingValidators[i])) {
+                i++;
+            } else {
+                confirmingValidators[i] = confirmingValidators[confirmingValidators
```

Comment:
I hope this is because of gas/storage optimization, because it's definitely not straightforward ;-)

Reply:
Do you mean deleting elements from the array?

Reply:
Yes. I see what the code does and it seems to work, it's just not that obvious. But rewriting the array is probably pretty costly. So 👍